Daviddt17/island-android-app
# Incremental Syncs

Addresses issue #72. 

Added incremental syncs using the `AlarmManager`. Incremental syncs were added to add system notifications without the user having to manually refresh. For now the default increment is 30 minutes.

Some notes on how this works:
- `SyncAlarm` is broadcasted to on every increment
- `SyncAlarm` calls `NotificationHelper.startListening()` and a sync is done subsequently.
- The `StopSystemNotificaionsReceiver` is broadcasted to when the sync is done. This is when `NotificationHelper.stopListening` is called.

Testing notifications while app is off:
- Set `SyncAlarm.SYNC_INTERVAL_MILLISECONDS` to something like 5000 (5 seconds)
- Make a post and close the app
- Comment on post with another device.
- This should cause the original device to add a system notification on the next sync.

The `BootReceiver` was added because alarms are deleted when the device is turned off. We should only have `BootReceiver` and `SyncAlarm` enabled when an account has actually been made. 

This PR does NOT address #74.
